compression/huffmancoding_test.go

```diff
@@ -0,0 +1,60 @@
+// huffmancoding_test.go
+// description: Tests the compression, encoding and decoding algorithms of huffmancoding.go.
+// author(s) [pedromsrocha](https://github.com/pedromsrocha)
+// see huffmancoding.go
+
+package compression
+
+import (
+	"sort"
+	"testing"
+)
+
+// The algorithm operates on []SymbolFreq sorted by frequency
+type ByFreq []SymbolFreq
+
+func (x ByFreq) Len() int           { return len(x) }
+func (x ByFreq) Less(i, j int) bool { return x[i].freq < x[j].freq }
+func (x ByFreq) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
```

Comment:
If you were to use the `sort.Slice` you don't need to implement the `sort.Interface` though I would much rather prefer if you use this repositories sorting functions - might have to modify them in a similar way to the standard library but I think it might also be possible to completely avoid it.

Reply:
Thanks for your suggestion of using sort.Slice instead, commit 4dc9b7d does that.

I think that the sorting functions in this repository's sort package apply to slices of constraints.Ordered elements, which are only basic datatypes (Int, String, Float), whereas in this case I need to sort a slice of pairs. I don't know which changes these would imply to the sort package.